src/library/scala/collection/immutable/Vector.scala

```diff
@@ -632,10 +638,11 @@ private final class Vector3[+A](_prefix1: Arr1, private[immutable] val len1: Int
     prepend1IfSpace(prefix1, prefix) match {
       case null => super.prependedAll0(prefix, k)
       case prefix1b =>
+        val diff = prefix1b.length - prefix1.length
         copy(prefix1 = prefix1b,
-             len1    = len1 - prefix1.length + prefix1b.length,
-             len12   = len12 - prefix1.length + prefix1b.length,
-             length0 = length0 - prefix1.length + prefix1b.length,
+             len1    = len1 + diff,
+             len12   = len12 + diff,
+             length0 = length0 + diff,
```

Comment:
I wasn't sure if the order of operations was to avoid overflow, so I did not factor out. But I didn't look closely either.

Reply:
That shouldn't be a problem here. The numbers are all well below `Int.MaxValue/2`. (`diff` is even <= 32)